src/java.desktop/share/classes/javax/imageio/stream/ImageInputStreamImpl.java

```diff
@@ -244,8 +243,8 @@ public short readShort() throws IOException {
             throw new EOFException();
         }
         return (byteOrder == ByteOrder.BIG_ENDIAN)
```

Comment:
This could just be `ByteArray.getShortBO(byteBuff, 0, byteOrder == ByteOrder.BIG_ENDIAN)`. Same for the others.